compose/service.py

```diff
@@ -477,9 +486,6 @@ def _next_container_number(self, one_off=False):
         return 1 if not numbers else max(numbers) + 1
 
     def _get_links(self, link_to_self):
-        if self.use_networking:
-            return []
-
```

Comment:
Ah yes. In fact, it still needs to be `if self.use_networking` - it's possible for a service in a v2 app to have no networks (by defining `networks: []`).